default/python/AI/AIstate.py

```diff
@@ -173,6 +172,41 @@ def __setstate__(self, state):
                     error("Could not find the aggression level of the AI, defaulting to typical.", exc_info=True)
                     aggression = fo.aggression.typical
             self.__init__(aggression)
+            return
+
+        from SaveGameManager import assert_content
+        assert_content(state, "version", int, False)
+        assert state["version"] == AIstate.version, "state version: %d, AIstate version: %d" % (
+            state["version"], AIstate.version)
+        assert_content(state, "uid", str, False)
+        assert_content(state, "turn_uids", dict, False)
+        for k, v in state["turn_uids"].iteritems():
+            assert type(k) == int, "k was %s" % k
+            assert type(v) == str, "v was %s" % v
+        assert_content(state, "_aggression", int, False)
+
+        for content in ("colonisablePlanetIDs", "colonisableOutpostIDs"):
```

Comment:
I would appreciate comments before blocks of code like this that summarize what they do